File: welcome/pages/welcome.go
package pages

import (
	"github.com/arthurmelton/KabaOS/welcome/window"
	"github.com/diamondburned/gotk4/pkg/gtk/v4"
)

func Welcome(w *window.Welcome, increase bool) {
	b := gtk.NewBox(gtk.OrientationVertical, 0)
	b.Append(gtk.NewLabel("(Logo)"))
	b.Append(gtk.NewLabel("Thank you for using KabaOS!"))

	w.Window.SetTitle("Welcome")

	w.Window.SetChild(b)
}

File: welcome/src/pages/welcome.zig
const window = @import("../main.zig").global_window;
const c = @cImport(@cInclude("gtk/gtk.h"));

pub fn page(_: bool) void {
    const box = c.gtk_box_new(c.GTK_ORIENTATION_VERTICAL, 0);
    c.gtk_box_append(@ptrCast(box), c.gtk_label_new("(Logo)"));
    c.gtk_box_append(@ptrCast(box), c.gtk_label_new("Thank you for using KabaOS!"));
    c.gtk_window_set_child(@ptrCast(window.window.?), box);

    c.gtk_window_set_title(@ptrCast(window.window.?), "Welcome");
}
